mne/io/curry/tests/test_curry.py

```diff
@@ -46,83 +48,83 @@
     do_warn = ''
 
 
+@pytest.fixture(scope='session')
+def bdf_curry_ref():
+    """Return a view of the reference bdf used to create the curry files."""
+    raw = read_raw_bdf(bdf_file, preload=True).drop_channels(['Status'])
+    return raw
+
+
 @pytest.mark.filterwarnings(do_warn)
 @testing.requires_testing_data
-def test_read_raw_curry():
+@pytest.mark.parametrize('fname', [
+    pytest.param(curry7_bdf_file, id='curry 7'),
+    pytest.param(curry8_bdf_file, id='curry 8'),
+])
+@pytest.mark.parametrize('preload', [True, False])
+def test_read_raw_curry(fname, preload, bdf_curry_ref):
     """Test reading CURRY files."""
-    # make sure all types of files can be passed to read_raw_curry
-    for filename in os.listdir(curry_dir):
-        if "test_bdf_stim_channel Curry" in filename:
-            read_raw_curry(op.join(curry_dir, filename))
+    raw = read_raw_curry(fname, preload=preload)
+
+    if preload:
+        isinstance(getattr(raw, '_data'), np.ndarray)
+    else:
+        with pytest.raises(AttributeError, match="no attribute '_data'"):
+            getattr(raw, '_data')
+
+    assert raw.n_times == bdf_curry_ref.n_times
+    assert raw.info['sfreq'] == bdf_curry_ref.info['sfreq']
+
+    for field in ['kind', 'ch_name']:
+        assert_array_equal([ch[field] for ch in raw.info['chs']],
+                           [ch[field] for ch in bdf_curry_ref.info['chs']])
 
-    # check no preloading for preload=False
-    with pytest.raises(AttributeError, match="no attribute '_data'"):
-        read_raw_curry(curry7_bdf_file, preload=False)._data
-        read_raw_curry(curry7_bdf_ascii_file, preload=False)._data
+    assert_allclose(raw.get_data(), bdf_curry_ref.get_data())
 
+    picks, start, stop = ["C3", "C4"], 200, 800
+    assert_allclose(
+        raw.get_data(picks=picks, start=start, stop=stop),
+        bdf_curry_ref.get_data(picks=picks, start=start, stop=stop),
+    )
+
+
+@pytest.mark.filterwarnings(do_warn)
+@testing.requires_testing_data
+@pytest.mark.parametrize('fname,tol', [
+    pytest.param(curry7_rfDC_file, 1e-6, id='curry 7'),
+    pytest.param(curry8_rfDC_file, 1e-3, id='curry 8'),  # XXX: why 1e-3?
```

Comment:
There are some values that turned out slightly different, maybe due to some floating point + rounding stuff? The data from these two files is read exactly the same, so i'd say this is the best we'd get of it...